lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -159,10 +159,14 @@ SPIRVToLLVMDbgTran::transCompilationUnit(const SPIRVExtInst *DebugInst) {
   M->addModuleFlag(llvm::Module::Max, "Dwarf Version", DWARFVersion);
   SPIRVWord SourceLang =
       getConstantValueOrLiteral(Ops, LanguageIdx, DebugInst->getExtSetKind());
-  if (DebugInst->getExtSetKind() == SPIRVEIS_NonSemantic_Shader_DebugInfo_200)
+  if (DebugInst->getExtSetKind() == SPIRVEIS_NonSemantic_Shader_DebugInfo_200) {
     SourceLang = convertSPIRVSourceLangToDWARFNonSemanticDbgInfo(SourceLang);
-  else
+  } else if (isSPIRVSourceLangValid(SourceLang)) {
     SourceLang = convertSPIRVSourceLangToDWARF(SourceLang);
+  } else {
+    M->addModuleFlag(llvm::Module::Warning, "Source Lang Literal", SourceLang);
```

Comment:
Can we think about unconditionally adding this info? In that case, we do not need to worry about adding and maintaining changes in spirv.debug.h.

This is a non-blocking comment.

Thanks

Reply:
> Can we think about unconditionally adding this info? In that case, we do not need to worry about adding and maintaining changes in spirv.debug.h.
> 
> This is a non-blocking comment.
> 
> Thanks

Do you mean whether we could always generate the "Source Lang Literal" metadata node without checking `isSPIRVSourceLangValid(SourceLang)`? 
I personally think that reducing the amount of metadata when possible makes most sense. I will try to rework this and get rid of the check once we have the new debug info spec. Ideally this could go in another pull request.